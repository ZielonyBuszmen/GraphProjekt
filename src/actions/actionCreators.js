import {INCREMENT_LIKES} from '../reducers/likesReducer';
import { BLOCK_GRAPH_INPUTS, CHECK_GRAPH_CONNECTIVITY } from '../reducers/connectivityReducer';
import { CREATE_EMPTY_MATRIX, LISTY_SASIEDZTWA_CHANGED, MACIERZ_SASIEDZTWA_CHANGED } from '../reducers/matrixReducer';
import { CHANGE_GRAPH_SIZE } from '../reducers/graphSizeRecuder';

export function increment(ktory, info) {
  return {
    type: INCREMENT_LIKES,
    ktory,
    info,
    dodatkowe: 'jakies inne dane',
  };
}

/**
 * @param graph - array[][], graf w formie macierzy sąsiedztwa
 * @return {{type, graph: *}}
 */
export function checkGraphConnectivity(graph) {
  return {
    type: CHECK_GRAPH_CONNECTIVITY,
    graph,
  }
}

/**
 * @param col - kolumna
 * @param row - wiersz
 * @param value - wartość
 * @returns {{type: string, col: *, row: *, value: *}}
 */
export function macierzSasiedztwaChanged(row, col, value) {
  return {
    type: MACIERZ_SASIEDZTWA_CHANGED,
    row,
    col,
    value,
  }
}

/**
 * @param size
 * @returns {{type: string, size: *}}
 */
export function createEmptyMatrix(size) {
  return {
    type: CREATE_EMPTY_MATRIX,
    size
  }
}

/**
 * @param lists - array[][], graf w formie listy sąsiedztwa
 * @return {{type, graph: *}}
 */
export function listySasiedztwaChanged(lists) {
  return {
    type: LISTY_SASIEDZTWA_CHANGED,
    lists
  }
}

/**
 * @param block - boolean
 * @returns {{type: string, block: boolean}}
 */
export function blockGraphInputs(block = true) {
  return {
    type: BLOCK_GRAPH_INPUTS,
    block
  }
}

export function graphSizeChanged(size) {
  return {
    type: CHANGE_GRAPH_SIZE,
    size
  }
}