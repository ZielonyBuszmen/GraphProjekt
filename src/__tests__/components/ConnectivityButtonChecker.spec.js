import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import '../../enzymeConfiguration';
import {
  ConnectivityButtonChecker,
  mapDispatchToProps,
  mapStateToProps,
} from '../../components/ConnectivityButtonChecker';
import { blockGraphInputs, checkGraphConnectivity } from '../../actions/actionCreators';

describe('ConnectivityButtonChecker', () => {
  it('should render component', () => {
    const props = {
      blockGraphInputs: () => {
      },
      checkGraphConnectivity: () => {
      },
      connectivity: false,
      matrix: [],
    };
    const component = shallow(<ConnectivityButtonChecker {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render success message', () => {
    const props = {
      blockGraphInputs: () => {
      },
      checkGraphConnectivity: () => {
      },
      connectivity: {
        blockGraphInputs: true,
        isGraphConnected: true,
      },
      matrix: [],
    };
    const component = shallow(<ConnectivityButtonChecker {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render fail message', () => {
    const props = {
      blockGraphInputs: () => {
      },
      checkGraphConnectivity: () => {
      },
      connectivity: {
        blockGraphInputs: true,
        isGraphConnected: false,
      },
      matrix: [],
    };
    const component = shallow(<ConnectivityButtonChecker {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should checkConnectivityClick', () => {
    const blockGraphInputs = (value) => {
      expect(value).toEqual(true);
    };
    const checkGraphConnectivity = (matrix) => {
      expect(matrix).toEqual(['matrix']);
    };

    const props = {
      blockGraphInputs,
      checkGraphConnectivity,
      connectivity: {},
      matrix: ['matrix'],
    };
    const component = shallow(<ConnectivityButtonChecker {...props} />);
    component.find('Button').simulate('click');
  });
});

describe('mapStateToProps', () => {
  it('should return object with state', () => {
    const state = {
      connectivity: false,
      matrix: ['matrix'],
    };
    const result = mapStateToProps(state);
    expect(result).toEqual(state);
  });
});

describe('mapDispatchToProps', () => {
  it('should contain dispatch', () => {
    const dispatch = arg => arg;
    const result = mapDispatchToProps(dispatch);

    expect(result.checkGraphConnectivity(['matrix']))
      .toEqual(checkGraphConnectivity(['matrix']));

    expect(result.blockGraphInputs())
      .toEqual(blockGraphInputs(true));
  });
});
