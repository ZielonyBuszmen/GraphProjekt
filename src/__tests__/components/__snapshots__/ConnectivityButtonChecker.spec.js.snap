// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectivityButtonChecker should render component 1`] = `
<div>
  <Button
    color="secondary"
    onClick={[Function]}
    size="lg"
    style={
      Object {
        "backgroundColor": "#6B0F19",
        "color": "#ffffff",
      }
    }
    tag="button"
  >
    Sprawdź silną spójność
  </Button>
</div>
`;

exports[`ConnectivityButtonChecker should render fail message 1`] = `
<div>
  <Alert
    className="alert-danger"
    closeAriaLabel="Close"
    color="success"
    isOpen={true}
    tag="div"
    transition={
      Object {
        "appear": true,
        "baseClass": "fade",
        "baseClassActive": "show",
        "enter": true,
        "exit": true,
        "in": true,
        "mountOnEnter": false,
        "onEnter": [Function],
        "onEntered": [Function],
        "onEntering": [Function],
        "onExit": [Function],
        "onExited": [Function],
        "onExiting": [Function],
        "tag": "div",
        "timeout": 150,
        "unmountOnExit": true,
      }
    }
  >
    <b>
      Graf nie jest silnie spójny.
    </b>
  </Alert>
  <Button
    color="secondary"
    onClick={[Function]}
    size="lg"
    style={
      Object {
        "backgroundColor": "#6B0F19",
        "color": "#ffffff",
      }
    }
    tag="button"
  >
    Sprawdź silną spójność
  </Button>
</div>
`;

exports[`ConnectivityButtonChecker should render success message 1`] = `
<div>
  <Alert
    className="alert-success"
    closeAriaLabel="Close"
    color="success"
    isOpen={true}
    tag="div"
    transition={
      Object {
        "appear": true,
        "baseClass": "fade",
        "baseClassActive": "show",
        "enter": true,
        "exit": true,
        "in": true,
        "mountOnEnter": false,
        "onEnter": [Function],
        "onEntered": [Function],
        "onEntering": [Function],
        "onExit": [Function],
        "onExited": [Function],
        "onExiting": [Function],
        "tag": "div",
        "timeout": 150,
        "unmountOnExit": true,
      }
    }
  >
    <b>
      Graf jest silnie spójny!
    </b>
  </Alert>
  <Button
    color="secondary"
    onClick={[Function]}
    size="lg"
    style={
      Object {
        "backgroundColor": "#6B0F19",
        "color": "#ffffff",
      }
    }
    tag="button"
  >
    Sprawdź silną spójność
  </Button>
</div>
`;
